compiler/main/driver.cpp

```diff
@@ -1974,6 +1974,12 @@ static void checkLLVMCodeGen() {
               "       'chpl' was built without LLVM support.  Either select a different\n"
               "       target compiler or re-build your compiler with LLVM enabled.");
 #endif
+  if (fLlvmCodegen) {
+    if ((ccflags.find("-pg") != std::string::npos) ||
+        (ldflags.find("-pg") != std::string::npos)) {
+      USR_WARN("The LLVM target compiler does not currently support '-pg'");
+    }
+  }
```

Comment:
The checks for `ccflags` could be put after `splitStringWhitespace(ccflags, clangCCArgs);` in `clangUtil` so that you have more precise checking that isn't subject to the substring issue you pointed out, but I don't there is an equivalent for `ldflags`. I did a quick search through the flag index (https://clang.llvm.org/docs/genindex.html) and didn't see anything that would conflict, so it should be good to just leave. (especially since we will hopefully support this better someday)

Reply:
Oh, that's interesting.  I'm not seeing anything that sloshes `ccFlags` into `llvmFlags`, though.  Unless this code does that after sloshing ccflags/ldflags into the compiler invocation's LLVM flags?  (I was only tracing backwards from llvmFlags, not forwards from ccflags).

```c
    // Start with any -mllvm options from the Clang invocation                  
    auto clangMLLVM = CI->getFrontendOpts().LLVMArgs;
    for (auto & arg : clangMLLVM) {
      vec.push_back(arg);
    }
```

But by that point, do we have any guarantees that something other than the user's options didn't add `-pg` to the list?  (i.e., wouldn't want to warn the user if their command-line was clean?  Or would we...maybe whatever that other thing was deserves a warning as well?)

Anyway, I'd be up for moving it if you're more confident about the meaning of `llvmFlags` than I am and would prefer it (where I'd probably just keep the same logic for the ldflags).

> I ... didn't see anything that would conflict

I was more worried about other user strings that might conflict, like if I added some flag like `-I /usr/include/pork-iz-pgs`

Reply:
> Anyway, I'd be up for moving it if you're more confident about the meaning of llvmFlags than I am and would prefer it (where I'd probably just keep the same logic for the ldflags).

I am not more confident in its meaning/usage :smile:, just what I found in a quick grep. `clangCCArgs` gets computed in `computeClangArgs`, which is then used to set the clangFlags in the dyno context and `runClang`. `runClang` that passes that list to `ClangInfo`. This seems like it would be another logical place to put the checks

Another, perhaps heavier handed approach is to use `std::regex` with something like `\B-pg\B`

Reply:
Oh, that's a really smart idea.  I suck at regex, though for anything non-trivial.  How confident are you in that pattern?  :)

Reply:
Maybe 90%? `std::regex_search(string, pat)` should do the trick I think

Reply:
Slight adjustment to the regex, using `-\bpg\b` should work.

Simple standalone test case in details

<details>

```c++
#include <regex>
#include <string>
#include <iostream>

int main() {
  auto r = std::regex("-\\bpg\\b");
  std::string s;

  s = "-I /usr/include/pork-iz-pgs h";
  std::cout << "matches '" << s << "': " << std::regex_search(s, r) << "\n";

  s = "-pg";
  std::cout << "matches '" << s << "': " << std::regex_search(s, r) << "\n";

  s = "foo -pg";
  std::cout << "matches '" << s << "': " << std::regex_search(s, r) << "\n";

  s = "-pg foo";
  std::cout << "matches '" << s << "': " << std::regex_search(s, r) << "\n";

  s = "foo -pg foo";
  std::cout << "matches '" << s << "': " << std::regex_search(s, r) << "\n";
}
```

</details>